tests/llm/utils/mock_toolset.py

```
@@ -875,10 +881,41 @@ def _configure_toolsets(
+                if toolset_mode == MockMode.LIVE or toolset_mode == MockMode.GENERATE:
+                    try:
+                        mcp_toolset.check_prerequisites()
+
+                        if (
+                            mcp_toolset.status != ToolsetStatusEnum.ENABLED
+                            and not self.allow_toolset_failures
+                        ):
+                            raise ToolsetPrerequisiteError(
+                                toolset_name=mcp_toolset.name,
+                                error_detail=mcp_toolset.error or "Unknown error",
+                            )
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)